### PR TITLE
fix(esl-utils): `@decorate` and `@bind` complete rework

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

File: src/modules/esl-utils/decorators/test/bind.test.ts [hunk 1]
```diff
@@ -1,73 +1,162 @@
...
+        const cb = instance.getThisName;
+        expect(cb()).toBe('1');
+      }
+    );
   });
 
-  test('inheritance basic binding test', () => {
-    class TestChildClass extends TestClass {
+  describe('Inheritance basic binding test', () => {
+    class TestPrent {
```
Comment (dshovchko) [inline src/modules/esl-utils/decorators/test/bind.test.ts:57]:
```suggestion
    class TestParent {
```

File: src/modules/esl-utils/decorators/test/bind.test.ts [hunk 2]
```diff
@@ -1,73 +1,162 @@
...
+      @bind getThis() {
+        return this;
+      }
+    }
+    class TestChild extends TestPrent {
```
Comment (dshovchko) [inline src/modules/esl-utils/decorators/test/bind.test.ts:62]:
```suggestion
    class TestChild extends TestParent {
```

File: src/modules/esl-utils/decorators/test/bind.test.ts [hunk 3]
```diff
@@ -1,73 +1,162 @@
...
+    instance2.postfix = '!';
+
+    test(
+      'Instances accessing superclass methods use corect context',
+      () => expect(instance2.getThisName()).toBe('Hello! world')
+    );
+    // test(
+    //   'Instances accessing superclass methods use exact unbounded version from prototype',
+    //   () => expect(instance2.getThisSuperNameMethod()).toBe(TestParent.prototype.getThisName)
+    // );
```
Comment (dshovchko) [inline src/modules/esl-utils/decorators/test/bind.test.ts:156]: commented out test

**Review by dshovchko — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

**Comment by ala-n** [comment]
Finally, now it checks the super and proto calls correctly...

**Comment by ala-n** [comment]
@exadel-inc/esl-core-team , @andreybelous, @nattallius , just in case you wondering where is the problem tests that I promised on the dev call...
I was sleepy enough to replace one condition with the same one but a little bit more obvious :) So everything was fine from the beginning )

**Review by dshovchko — APPROVED** [review]

**Review by andreybelous — APPROVED** [review]

File: src/modules/esl-utils/decorators/bind.ts [hunk 4]
```diff
@@ -1,47 +1,33 @@
...
     },
-    set(value: T): void {
+    set(value: Fn): void {
       Object.defineProperty(this, propertyKey, {
-        writable: true,
-        enumerable: false,
+        value,
+        writable: descriptor.writable,
+        enumerable: descriptor.enumerable,
         configurable: true,
```
Comment (yadamskaya) [inline src/modules/esl-utils/decorators/bind.ts:30]: minor (and the same in `decorate.ts`)
```suggestion
        configurable: true
```

**Review by yadamskaya — APPROVED** [review]

**Review by abarmina — APPROVED** [review]

**Pull request merged**